pkg/test/echo/response/check/checker.go

```diff
@@ -0,0 +1,46 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package check
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/test/echo/response"
+	"istio.io/istio/pkg/test/echo/response/checkcall"
+)
+
+// Checker inspects echo responses for errors.
+type Checker interface {
+	checkcall.Checker
+
+	// Check a single response.
+	Check(*response.Instance) error
+}
+
+// Func is a Checker implemented by only the Check function.
+type Func func(r *response.Instance) error
+
+func (c Func) Check(r *response.Instance) error {
+	return c(r)
+}
+
+func (c Func) CheckCall(rs response.Instances, _ error) error {
+	for i, r := range rs {
+		if err := c.Check(r); err != nil {
+			return fmt.Errorf("response[%d]: %v", i, err)
```

Comment:
is multierror useful or are we intentionally short-circuiting? seems like it could be useful to know 1 request failed vs all of them did?